services/venice-controller/src/main/java/com/linkedin/venice/controller/DeadStoreStatsPreFetchTask.java

```diff
@@ -40,32 +39,47 @@ public DeadStoreStatsPreFetchTask(String clusterName, VeniceHelixAdmin admin, lo
   public void run() {
     logger.info("Started {}", taskId);
     isRunning.set(true);
-    try {
-      // wait for 60 seconds for controller to become leader as it's required to be leader for dead store stats
-      long deadline = System.currentTimeMillis() + TimeUnit.SECONDS.toMillis(60);
-      while (!admin.isLeaderControllerFor(clusterName)) {
-        if (System.currentTimeMillis() > deadline) {
-          throw new VeniceException("Timed out waiting for controller to become leader for cluster: " + clusterName);
-        }
-        Utils.sleep(10_000); // sleep for 10 seconds
-      }
-
-      logger.debug("Initial fetch of dead store stats for cluster: {}", clusterName);
-      admin.preFetchDeadStoreStats(clusterName, getStoresInCluster());
-    } catch (Exception e) {
-      logger.error("Error during initial fetch of dead store stats for cluster: {}", clusterName, e);
-    }
 
     while (isRunning.get()) {
       try {
-        Utils.sleep(refreshIntervalMs);
-        long startTime = System.currentTimeMillis();
-        logger.debug("Fetching dead store stats for cluster: {}", clusterName);
-        admin.preFetchDeadStoreStats(clusterName, getStoresInCluster());
-        logger.info("Successfully refreshed dead store stats in {} ms", System.currentTimeMillis() - startTime);
+        boolean isCurrentlyLeader = admin.isLeaderControllerFor(clusterName);
+
+        // Handle leadership state transitions
+        if (!isCurrentlyLeader && wasLeaderInPreviousCycle) {
+          logger.info(
+              "Controller is no longer leader for cluster: {}. Stopping dead store stats pre-fetch task.",
+              clusterName);
+          wasLeaderInPreviousCycle = false;
+          break;
+        } else if (isCurrentlyLeader && !wasLeaderInPreviousCycle) {
+          // Newly became leader - immediate prefetch without waiting for refresh interval
+          logger.info(
+              "Controller newly became leader for cluster: {}. Triggering immediate dead store stats prefetch.",
+              clusterName);
+          wasLeaderInPreviousCycle = true;
+
+          long startTime = System.currentTimeMillis();
+          admin.preFetchDeadStoreStats(clusterName, getStoresInCluster());
```

Comment:
can we reduce the code duplication? I am seeing that if `isCurrentlyLeader` is true, then we always run `admin.preFetchDeadStoreStats`